### PR TITLE
[WIP] Use external prop for BlockEditor in test

### PR DESCRIPTION
**Pull request by CaoRuiming**
Getting methods and objects through the external prop seems to work fine, but for some reason `DropNodeTarget` in `DragAndDropEditor` seems to prevent the node from constructing/mounting and setting `this.cm` in `BlockEditor`.

### CODE REVIEW

File: src/ui/BlockEditor.js [hunk 1]
```diff
@@ -299,6 +299,8 @@ class BlockEditor extends Component {
 
     ed.on('changes', this.editorChange);
 
+    this.props.external.cm = ed;
+    this.cm = ed;
```
Comment (schanzer) [inline src/ui/BlockEditor.js]: Do we need `this.cm`?

File: src/ui/Search.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import {Tab, Tabs, TabList, TabPanel} from 'react-tabs';
 import 'react-tabs/style/react-tabs.less';
 
 export default (Editor, searchModes) => {
+  Modal.setAppElement(document.createElement('div'));
```
Comment (schanzer) [inline src/ui/Search.js]: Why do we need this? See `componentDidMount`, below.

File: src/ast.js [hunk 3]
```diff
@@ -90,6 +117,88 @@ export class AST {
...
+    // If we have a DOM elt, use it and update the id. Mark parents of nodes with DOM elts as dirty
+    newAST.nodeIdMap.forEach(n => { if(!n.el){ dirtyNodes.add(newAST.getNodeParent(n) || n); }});
+    // Ensure that no dirty node is the ancestor of another dirty node
+    let dirty = [...dirtyNodes].sort((a, b) => a.path<b.path? -1 : a.path==b.path? 0 : 1);
+    dirty.reduce((n1, n2) => n2.path.includes(n1.path)? dirtyNodes.delete(n2) && n1 : n2, false);
+    newAST.dirtyNodes = new Set([...dirtyNodes].map(n => newAST.getNodeByPath(n.path)) // grab all the nodes
+      .filter(n => n !== undefined));                                                  // remove deleted ones
+    return newAST;
+  }
+
```
Comment (schanzer) [inline src/ast.js]: Where is all this coming from? Maybe leftover from basing off `new-reactify-with-tests`?  This file shouldn't have any of these changes. It should be reset the to what it is on `new-reactify`.

File: example/new-editor-example.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import wescheme from '../src/languages/wescheme';
+import 'codemirror/lib/codemirror.css';
```
Comment (schanzer) [inline example/new-editor-example.js]: Why is this needed? It's already important from within ToggleEditor. I suspect this is leftover from my old `-with-tests` branch, and can be safely deleted

File: src/CodeMirrorBlocks.js [hunk 5]
```diff
@@ -6,21 +6,13 @@ import wescheme from '../src/languages/wescheme';
...
-      />,
-      container
+      <ToggleEditor language={language} initialCode={code} external={obj} />,
+      dom
     );
     return obj;
   }
 }
-
-module.exports = CodeMirrorBlocks;
```
Comment (schanzer) [inline src/CodeMirrorBlocks.js]: This file is also out of date. The one on `new-reactify` is correct

File: spec/index.js [hunk 6]
```diff
@@ -1,8 +1,9 @@
 // *Some* environments (phantomjs) don't have es5 (Function.prototype.bind)
-require('babel-polyfill');
+require('@babel/polyfill');
+// require('babel-polyfill');
 
 // require all the files in the spec folder that end with -test.js
-var context = require.context('.', true, /.-test.js$/);
+var context = require.context('.', true, /.new-blocks-test.js$/);
```
Comment (schanzer) [inline spec/index.js]: Confirming that you wanted to turn off ALL tests except for `new-blocks-test`?

Comment (CaoRuiming) [reply spec/index.js]: This is a temporary measure so I don't get overwhelmed by the error messages from the outdated tests while I work on new-blocks-test

File: example/new-editor-example.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import wescheme from '../src/languages/wescheme';
-import CodeMirrorBlocks from '../src/CodeMirrorBlocks';
+import {CodeMirrorBlocks} from '../src/CodeMirrorBlocks';
```
Comment (justinpombrio) [inline example/new-editor-example.js]: Since you updated `src/CodeMirrorBlocks` to match `new-reactify`, this import needs to be `import CodeMirrorBlocks` rather than `import {CodeMirrorBlocks}`. Same thing in `spec/new-block-test`.
